dashboard/views.py

```diff
@@ -154,6 +154,45 @@ def get_queryset(self) -> QuerySet[Achievement]:
 		).order_by('-obtained', '-num_found')
 
 
+class AchievementLeaderboard(
+	LoginRequiredMixin, SuccessMessageMixin, UpdateView[Student, BaseModelForm[Student]]
+):
+	model = Student
+	fields = ('public', )
+	success_url = reverse_lazy('achievement-leaderboard')
+	object: Student
+	template_name = 'dashboard/achievement_leaderboard.html'
+
+	def form_valid(self, form: BaseModelForm[Student]):
+		messages.success(self.request, f"Updated settings for {self.object.user.username}!")
+		form.instance.user = self.request.user
+		return super().form_valid(form)
+
+	def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+		context = super().get_context_data(**kwargs)
+		context['rows'] = self.get_queryset()
+		return context
+
+	def get_queryset(self) -> List[Dict[str, Any]]:
+		students = Student.objects.filter(semester__active=True)
+		rows = get_student_rows(students)
+		rows.sort(key=lambda row: (
+			-row['diamonds'],
+			row['student'].name.upper(),
+		))
+		return rows[0:25]
+
+	def get_object(self, queryset: QuerySet[Student] = None) -> Student:
+		student = Student.objects.get(user=self.request.user)
```

Comment:
This line is probably the biggest issue. `Student.objects.get(...)` will expect there to be exactly one return object. So someone who is logged in with no student account (e.g. me) will get a 500 here. Similarly, returners will get a 500 here because they have > 1 student object attached.

You can fix the latter issue by also requiring `semester__active=True` but I'd have to think about what to do with the former: in that case there should be no form, so using `UpdateView[Student, BaseModelForm[Student]]` is sort of painting ourselves into a corner.

Honestly it might be easier to ditch the class-based view and do things by hand here if the form is going to be optional? idk.